### PR TITLE
Add a chart that groups anomaly rate by chart type.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

Adds a new chart that groups the anomaly rate of dimensions based on the chart's type.

##### Test Plan

- CI jobs
- ML VMs

### CODE REVIEW

**Comment by andrewm4894** [comment]
looking good, wondering if we should change `anomaly_detection.anomaly_rate` to be `anomaly_detection.node_anomaly_rate` to be consistent with convention of `anomaly_detection.type|user|app_anomaly_rate` that this new approach suggests.

image

File: ml/ad_charts.cc [hunk 1]
```diff
@@ -258,6 +258,56 @@ void ml_update_host_and_detection_rate_charts(ml_host_t *host, collected_number
...
+            snprintfz(id_buf, 1024, "type_anomaly_rate_on_%s", localhost->machine_guid);
+            snprintfz(name_buf, 1024, "type_anomaly_rate_on_%s", rrdhost_hostname(localhost));
+
+            host->type_anomaly_rate_rs = rrdset_create(
+                    host->rh,
+                    "anomaly_detection", // type
+                    id_buf, // id
+                    name_buf, // name
+                    "anomaly_rate", // family
+                    "anomaly_detection.anomaly_rate", // ctx
```
Comment (andrewm4894) [inline ml/ad_charts.cc:278]:
```suggestion
                    "anomaly_detection.type_anomaly_rate", // ctx
```

I think a seperate context makes sense such that ultimatly we would have below contexts:

- anomaly_detection.node_anomaly_rate
- anomaly_detection.type_anomaly_rate
- anomaly_detection.app_anomaly_rate
- anomaly_detection.user_anomaly_rate

**Comment by andrewm4894** [comment]
Looking at this again i'm fairly sure we need a new chart for `anomaly_detection.type_anomaly_rate` so that we end up with the existing `anomaly_detection.anomaly_rate` chart (which we should leave alone since we don't want to impact any anomaly advisor stuff) and then below that we want the `anomaly_detection.type_anomaly_rate`. 

So from `anomaly_detection.anomaly_rate` you can see a summary line as a quick glace:

image

and then just below that we want `anomaly_detection.type_anomaly_rate` chart that will look like this so you can see what lower level stuff had elevated anomaly rates:

image

- `anomaly_rate`  dimension on `anomaly_detection.anomaly_rate` is always going to be naturally lower than individual dims on `anomaly_detection.type_anomaly_rate` so it gets lost when on same chart.
- eventually we will want `anomaly_detection.app_anomaly_rate` and `anomaly_detection.user_anomaly_rate` as two seperate charts too.
- splitting out like this also means we have no impact on anomaly advisor in any way. whereas at moment we do - see chart below:
image

So while was not sure at first i'm 99% sure now that i think we need to split it out into its own `anomaly_detection.type_anomaly_rate` context/chart.

**Comment by andrewm4894** [comment]
will drop and recreate my nodes to check the charts split as expected.

**Comment by andrewm4894** [comment]
working as expected - will let it run for rest of night and check it some more tommorow.

image

**Review by andrewm4894 — APPROVED** [review]
works as expected

**Comment by andrewm4894** [comment]
wondering about if type chart should be line (as-is):

image

or area (a bit prettier in some way) :

image

or stacked (lets you see more clearly what stuff was overlapping and when):

image

i think i like stacked but don't want to derail this PR - we can easy debate and change it later if we want in a different PR.

**Review by thiagoftsm — APPROVED** [review]
After @andrewm4894's comment and a local test, LGTM!

**Comment by vkalintiris** [comment]
> i think i like stacked but don't want to derail this PR

I prefer stacked too, I'll open a new PR.

**Pull request merged**